controllers/ocirepository_controller.go

```diff
@@ -942,3 +971,64 @@ func (r *OCIRepositoryReconciler) notify(ctx context.Context,
 		}
 	}
 }
+
+// notify emits notification related to the reconciliation.
+func (r *OCIRepositoryReconciler) verify(ctx context.Context, obj *sourcev1.OCIRepository, url string) (bool, error) {
+	// get the public keys from the given secret
+	secretRef := obj.Spec.Verify.SecretRef
+
+	ref, err := name.ParseReference(url)
+	if err != nil {
+		return false, err
+	}
+
+	if secretRef != nil {
+		certSecretName := types.NamespacedName{
+			Namespace: obj.Namespace,
+			Name:      secretRef.Name,
+		}
+
+		var pubSecret corev1.Secret
+		if err := r.Get(ctx, certSecretName, &pubSecret); err != nil {
+			return false, err
+		}
+
+		// traverse all public keys and try to verify the signature
+		// this is brute-force approach, but it is ok for now
+		for _, data := range pubSecret.Data {
+			pubRaw, err := base64.StdEncoding.DecodeString(string(data))
+			if err != nil {
+				return false, err
```

Comment:
This error all the others must be wrapped with `fmt.Errorf` and they should contain helpful messages for end-users, things like the secret name, key, which step failed, etc.

Reply:
we had to delete this because we don't need to decode the string since client-go gives the data of the secret already decoded.